File: sources/RedStar 2.0/me/utils/fluxfont/FontManager.java
package me.utils.fluxfont;

import java.awt.Font;
import java.io.InputStream;
import me.yarukon.font.Yarukon;

public class FontManager {
    public static Font getFont(String name, int size) {
        Font font;
        try {
            InputStream is = FontManager.class.getResourceAsStream("/assets/minecraft/pride/font/" + name);
            font = Font.createFont(0, is);
            font = font.deriveFont(0, size);
        }
        catch (Exception ex) {
            System.out.println("Error loading font " + name);
            font = new Font("arial", 0, size);
        }
        return font;
    }

    static {
        new Yarukon();
    }
}

File: sources/Leaf devEnv/com/leafclient/leaf/utils/render/font/FontManager.kt
package com.leafclient.leaf.utils.render.font

import java.awt.Font

object FontManager {

    private val fontCache = mutableMapOf<FontData, TTFFontRenderer>()

    /**
     * Allows renderers to create a [TTFFontRenderer], this method
     * creates the [TTFFontRenderer] if it hasn't been created yet.
     */
    operator fun get(name: String, size: Float)
        = fontCache.getOrPut(FontData(name, size * 2F)) {
        TTFFontRenderer(
                javaClass.getResourceAsStream("/assets/minecraft/font/$name.ttf")?.let {
                    Font.createFont(Font.TRUETYPE_FONT, it).deriveFont(size * 2F)
                } ?: Font(name, Font.TRUETYPE_FONT, (size * 2F).toInt())
            )
        }

    /**
     * A data class used to manage our fonts
     */
    data class FontData(val name: String, val size: Float)

}